core/src/test/java/org/elasticsearch/common/xcontent/builder/XContentBuilderTests.java

```diff
@@ -260,4 +263,36 @@ public void testCopyCurrentStructure() throws Exception {
 
         assertThat(i, equalTo(terms.size()));
     }
+
+    @Test
+    public void testHandlingOfPath() throws IOException {
+        Path path = PathUtils.get("path");
```

Comment:
I disagree on this one. We test on both Windows and multiple Linux distributions, which should cover us in terms of integration problems and this is exactly why I want to see the test.

In terms of getting an absolute path, you can call `createTempDir().toAbsolutePath()`, which should return a `Path` object to a temporary directory that you can safely use for absolute path testing in an OS agnostic way.